src/System.ServiceModel.Primitives/tests/Description/OperationBehaviorTest.cs

```diff
@@ -87,6 +90,37 @@ public static void XmlSerializerOperationBehavior_BasicUsage()
         }
     }
 
+    [WcfFact]
+    public static void DataContractSerializationSurrogateTest()
+    {
+        OperationDescription od = null;
+        DataContractSerializerOperationBehavior behavior = new DataContractSerializerOperationBehavior(od);
+
+        behavior.SerializationSurrogateProvider = new MySerializationSurrogateProvider();
+
+        DataContractSerializer dcs = (DataContractSerializer)behavior.CreateSerializer(typeof(SurrogateTestType), nameof(SurrogateTestType), "ns", new List<Type>());
+
+        var members = new NonSerializableType[2];
+        members[0] = new NonSerializableType("name1", 1);
+        members[1] = new NonSerializableType("name2", 2);
+
+        using (MemoryStream ms = new MemoryStream())
+        {
+            SurrogateTestType obj = new SurrogateTestType { Members = members };
+
+            dcs.WriteObject(ms, obj);
+            ms.Position = 0;
```

Comment:
DCS is very good at serializing types which have no attributes on them. I suspect this test would work if you didn't set the surrogate provider. Can you add a bool field to the provider which gets set to true when the provider is called and then verify that it's true at the end of the test?

Reply:
@mconnew If I comment out line 99, the test will fail with an `InvalidDataContractException: Type 'NonSerializableType' cannot be serialized. Consider marking it with the DataContractAttribute attribute...`

Reply:
Good to know. It would still be nice to have an affirmative unambiguous check to know that the class has been used.